### PR TITLE
Fix #109

### LINKED ISSUE #109: [wf-dock] Apply auto hide on startup (feature request)
If I set auto-hide it works as expected, but only after my mouse has hovered over it once. On login it appears on all displays (as expected) but remains visible until I've moved my mouse over it on each display. 

I'd like it to either start hidden  or start visible and auto-hide after the specified interval regardless of whether I hover my mouse over it.

### PR DESCRIPTION
**Pull request by NamorNiradnug**
fixes #109

### CODE REVIEW

**Review by ammen99 — COMMENTED** [review]
I am confused as to how this works and why it is necessary, would you mind elaborating?

**Comment by NamorNiradnug** [comment]
> I am confused as to how this works and why it is necessary, would you mind elaborating?

The reason the panel won't hide on startup is that [this](https://github.com/WayfireWM/wf-shell/blob/master/src/util/wf-autohide-window.cpp#L91) check fails, because `autohide_counter` is initialized with zero. So, as I understand what `autohide_counter` is, it should be incremented.

**Comment by NamorNiradnug** [comment]
Oh, I see the problem. It completely ignores if autohide is enabled or not. Maybe then `autohide_counter` should be incremented only if autohide is enabled?

**Comment by ammen99** [comment]
Your new approach certainly makes more sense :) I'm wondering whether it would make sense to instead make the generic autohide window handle the autohide options as well? I don't think the panel or the dock care about the specific value of the option, so we could share the code, do you think this is feasible?

**Comment by NamorNiradnug** [comment]
There are two problems currently
1. `wf-dock` is shown on the left instead of being centred when its `autohide` option is `false` and `exclusive_zone` is set
2. `wf-panel` hides automatically when a view on the corresponding output goes fullscreen. Shouldn't the feature also be shared between panel and dock as well?

**Comment by ammen99** [comment]
> 1. `wf-dock` is shown on the left instead of being centred when its `autohide` option is `false` and `exclusive_zone` is set

Hmm, sounds like this could be a bug on Wayfire's side. I'll try to look into it tomorrow.



> 2\. `wf-panel` [hides automatically](https://github.com/WayfireWM/wf-shell/blob/master/src/panel/panel.cpp#L307) when a view on the corresponding output goes fullscreen. Shouldn't the feature also be shared between panel and dock as well?

Yes, sounds good. IIRC the dock used to always auto-hide, which is why the feature wasn't necessary before.

File: src/util/wf-autohide-window.cpp [hunk 1]
```diff
@@ -50,6 +50,15 @@ WayfireAutohidingWindow::WayfireAutohidingWindow(WayfireOutput *output,
             if (this->active_button)
                 unset_active_popover(*this->active_button);
         });
+
+    if (output->output)
+    {
+        const auto* listener = new zwf_output_v2_listener {
+            .enter_fullscreen = [](void* data, zwf_output_v2 *){ ((WayfireAutohidingWindow*)data)->increase_autohide(); },
```
Comment (ammen99) [inline src/util/wf-autohide-window.cpp:57]: Woah, didn't know you can do that (passing lambda to wayland callbacks), cool :)

**Comment by ammen99** [comment]
> Hmm, sounds like this could be a bug on Wayfire's side. I'll try to look into it tomorrow.

I found the problem, Wayfire's anchored areas do not center the surface properly with exclusive zone. I have a few other changes planned in that respect, so I'll create a Wayfire PR soon.

**Comment by ammen99** [comment]
@NamorNiradnug https://github.com/WayfireWM/wayfire/pull/1829 should fix the issue with the dock placement, can you confirm?

**Review by ammen99 — APPROVED** [review]
LGTM, should we merge?

**Comment by NamorNiradnug** [comment]
There is a memory leak, because `listener` is never deleted, I'll fix it soon, please don't merge it yet.

**Comment by NamorNiradnug** [comment]
> @NamorNiradnug [WayfireWM/wayfire#1829](https://github.com/WayfireWM/wayfire/pull/1829) should fix the issue with the dock placement, can you confirm?

Yep, it works!

**Comment by NamorNiradnug** [comment]
> LGTM, should we merge?

I think it's ready to be merged now.

**Comment by ammen99** [comment]
Thanks!

**Pull request merged**